R/eic-report.R

```diff
@@ -0,0 +1,140 @@
+
+get_gh_token <- function (token = "") {
+
+    tryCatch (
+        gitcreds::gitcreds_get ()$password,
+        error = function (e) ""
+    )
+}
+
+get_issues_qry <- function (gh_cli,
+                            org = "ropensci",
+                            repo = "software-review",
+                            end_cursor = NULL) {
+
+    after_txt <- ""
+    if (!is.null (end_cursor)) {
+        after_txt <- paste0 (", after:\"", end_cursor, "\"")
+    }
+
+    q <- paste0 ("{
+        repository(owner:\"", org, "\", name:\"", repo, "\") {
+                   issues (first: 100", after_txt, ", states: [OPEN]) {
+                       pageInfo {
+                           hasNextPage
+                           endCursor
+                       }
+                       edges {
+                           node {
+                               ... on Issue {
+                                   number
+                                   createdAt
+                                   lastEditedAt
+                                   updatedAt
+                                   assignees (first: 100) {
+                                       nodes {
+                                           name
+                                       }
+                                   }
+                                   title
+                                   labels (first: 100) {
+                                       edges {
+                                           node {
+                                               name
+                                           }
+                                       }
+                                   }
+                                   url
+                                   comments (last: 100) {
+                                       nodes {
+                                           createdAt,
+                                           author {
+                                               login
+                                           },
+                                           body
+                                       }
+                                   }
+                               }
+                           }
+                       }
+                   }
+                }
+        }")
+
+    return (q)
+}
+
+#' Apply the query above to extract data from GitHub GraphQL API, and
+#' post-process into a `data.frame`.
+#' @noRd
+devguide_eic_gh_data <- function () {
+
+    token <- get_gh_token ()
+    gh_cli <- ghql::GraphqlClient$new (
```

Comment:
why not use `gh::gh_gql()`? (just curious)

Reply:
No real reason; i just copy-pasta'd code from elsewhere to get started. I've now changed, and was suprised at the effort it took. I think `ghql` does a much better job of hiding the ugliness of nested list nightmares, simply through naming everything. `gh` is admittedly more directly representative of a query, but it then leaves crucial parts unnamed, and thus only accessible through loads of `apply`-type calls that `ghql` does not require. I never really appreciated that difference before.

And the named/unnamed behaviour reflects this:
https://github.com/ropensci-org/devguider/blob/327eebb4dd33df2d856f33fed3b25bfdb9a95eff/R/eic-report.R#L29
which somehow translates in `ghql` to nothing at all, and so named list items simply jump to everything within there, whereas `gh` makes each `... on Issue` item a distinct *unnamed* list item. Which is annoying.